### ISSUE TITLE
support request to manage virtual machines in VDI format Virtualbox

### ISSUE THREAD

**Issue by maravento** [cockpit-project/cockpit-machines#877]

**Comment by jelly** [comment]
All machines created with the vbox libvirt driver should show up in cockpit as well if they show up in `virsh list`, do they not show up?

Or is the question how one migrates it's virtualbox Virtual machines to cockpit-machines? This seems to work fine for me with the `Import vm` feature.

**Comment by maravento** [comment]
You are right. I already did the tests. Cockpit allows it to import virtual disks in `.vdi` format. 
`virsh list` command does not show information about virtualbox VMs. I assume i have to import them first or tell the program where they are (I don't know how to do this part)
I would like to know too which are the recommended packages to install for VMs Virtualbox and similar. Summarizing what they explain in several places, it is:
`sudo apt install libvirt-daemon libvirt-daemon-system libvirt-daemon-driver-vbox`
Are these packages necessary?

**Comment by jelly** [comment]
> PD: virsh list command does not show information about virtualbox VMs. I assume i have to import them first or tell the program where they are (I don't know how to do this part)

So I was wrong here, this is expected, we won't show virtualbox Virtual machines in Cockpit-machines.

> Question: What are the recommended packages to install, related to virtualization: (Do I have to open another issue for this question?)
> Question: What are the recommended packages to install, related to virtualization: (Do I have to open another issue for this question?)


> @jelly You are right. I already did the tests. Cockpit allows it to import virtual disks in `.vdi` format (but it doesn't recognize the virtualbox VM files. I have to create a new VM with the `.vdi` disk). But for me it's fine. 

Right, a `vdi` file is different from a `vmdk` file. If you want to switch to `cockpit-machines` it's probably the best to convert the `vdi` file to `qcow2` or a `raw` image file. Which seems to be as easy as `qemu-img convert -f vdi -O qcow2 ubuntu.vdi ubuntu.qcow2`.

> PD: `virsh list` command does not show information about virtualbox VMs. I assume i have to import them first or tell the program where they are (I don't know how to do this part)

Right, that's because it connects to the user or system session. And not to vbox://, this will also not be supported in cockpit-machines. 

> Question: What are the recommended packages to install, related to virtualization: (Do I have to open another issue for this question?)
>  I install the following: `sudo apt install cockpit-machines virt-viewer` 

This is fine if you want to use `cockpit-machines`.

> But other sites say that i have to install the following additional packages: `sudo apt install libvirt-daemon libvirt-daemon-system libvirt-daemon-driver-vbox virt-manager` Which of these additional packages do you recommend? (or none)

If you want to use `virt-manager` then this is what you want. Just install cockpit-machines should pull in the required dependencies to start a virtual machine.

**Comment by maravento** [comment]
thanks for the clarifications. excellent program. That's all for now
